Platforms/Export-Import-Platform.ps1

```diff
@@ -110,136 +129,214 @@ Function Join-ExceptionMessage
+	#endregion
+
+	switch ($PsCmdlet.ParameterSetName) {
+		"Import" {
+			import-platform $PlatformZipPath -error
+		}
+
+		"ImportFile" {
+			foreach ($line in Get-Content $listFile) {
+				if ("" -ne $line) { import-platform $line }
+			} 
+		}
 
-	switch($PsCmdlet.ParameterSetName)
-	{
-		"Import"
-		{
-			If (Test-Path $PlatformZipPath)
-			{
-				$zipContent = [System.IO.File]::ReadAllBytes($(Resolve-Path $PlatformZipPath))
-				$importBody = @{ ImportFile=$zipContent; } | ConvertTo-Json -Depth 3 -Compress
-				try{
-					$ImportPlatformResponse = Invoke-RestMethod -Method POST -Uri $URL_ImportPlatforms -Headers $logonHeader -ContentType "application/json" -TimeoutSec 2700 -Body $importBody
-					Write-Debug "Platform ID imported: $($ImportPlatformResponse.PlatformID)"
-					Write-Host "Retrieving Platform details"
-					# Get the Platform Name
-					$platformDetails = Invoke-RestMethod -Method Get -Uri $($URL_PlatformDetails -f $ImportPlatformResponse.PlatformID) -Headers $logonHeader -ContentType "application/json" -TimeoutSec 2700
-					If($platformDetails)
-					{
-						Write-Debug $platformDetails
-						Write-Host "$($platformDetails.Details.PolicyName) (ID: $($platformDetails.PlatformID)) was successfully imported and $(if($platformDetails.Active) { "Activated" } else { "Inactive" })"
-						Write-Host "Platform details:" 
-						$platformDetails.Details | Select-Object PolicyID, AllowedSafes, AllowManualChange, PerformPeriodicChange, @{Name = 'AllowManualVerification'; Expression = { $_.VFAllowManualVerification}}, @{Name = 'PerformPeriodicVerification'; Expression = { $_.VFPerformPeriodicVerification}}, @{Name = 'AllowManualReconciliation'; Expression = { $_.RCAllowManualReconciliation}}, @{Name = 'PerformAutoReconcileWhenUnsynced'; Expression = { $_.RCAutomaticReconcileWhenUnsynched}}, PasswordLength, MinUpperCase, MinLowerCase, MinDigit, MinSpecial 
-					}		
-				} catch {
-					#Write-Error $_.Exception
-					Write-Error $_.Exception.Response
-					Write-Error $_.Exception.Response.StatusDescription
-				}
-			}
+		"Export" {
+			export-platform $PlatformID
 		}
-		"Export"
-		{
-			try{
-				$exportURL = $URL_ExportPlatforms -f $PlatformID
-				Invoke-RestMethod -Method POST -Uri $exportURL -Headers $logonHeader -ContentType "application/zip" -TimeoutSec 2700 -OutFile $PlatformZipPath 
-			} catch {
-				Write-Error $_.Exception.Response
-				Write-Error $_.Exception.Response.StatusDescription
-			}
+
+		"ExportFile" {
+			foreach ($line in Get-Content $listFile) {
+				$line 
+				if ("" -ne $line) { export-platform $line }		
+			} 
+	
+		}
+		"ExportActive" {
+			$platforms = Get-PlatformsList
+			$null | Out-File -FilePath "$PlatformZipPath\_Export.txt" -Force
+			foreach ($line in $platforms) {
+				$line 
+				("$PlatformZipPath\$line.zip").Replace("\\","\").Replace("/","\") | Out-File -FilePath "$PlatformZipPath\_Export.txt" -Append
+				if ("" -ne $line) { export-platform $line }		
+			} 
+
+		}
+		"ExportAll" {
```

Comment:
This is a duplication of the same code that you created in ExportActive
You can combine them in the Switch case like this:
{"ExportActive" -or "ExportAlll" } {
     ...
}
Then you can have a single code part and just use the switch of the Get-PlatformsList accordingly
$platforms - Get-PlatformsList -GetAll:$(($PsCmdlet.ParameterSetName -eq "ExportAll"))
OR
$platforms - Get-PlatformsList -ActiveOnly:$(($PsCmdlet.ParameterSetName -eq "ExportActive"))